auth.c

```diff
@@ -53,13 +68,95 @@ void janus_auth_deinit(void) {
 	if(allowed_plugins != NULL)
 		g_hash_table_destroy(allowed_plugins);
 	allowed_plugins = NULL;
+	if(auth_secret != NULL)
+		g_free(auth_secret);
+	auth_secret = NULL;
 	janus_mutex_unlock(&mutex);
 }
 
+gboolean janus_auth_check_signature(const char *token, const char *realm) {
+	if (!auth_enabled || auth_secret == NULL)
+		return FALSE;
+	gchar *token_signature = strchr(token, ':') + 1;
+	gchar *token_metadata = g_strndup(token, token_signature - token - 1);
+	gchar **list = g_strsplit(token_metadata, ",", 3);
+	/* need at least an expiry timestamp and realm */
+	if(!list[0] || !list[1])
+		goto fail;
+	/* verify timestamp */
+	int expiry_time = atoi(list[0]);
+	int real_time = janus_get_real_time() / 1000000;
+	if(real_time > expiry_time)
```

Comment:
Wouldn't this fail is application server and Janus don't share the same timezone, or simply cause issues if they don't share the same clock?

Reply:
yes, I am not sure about the guarantees that `janus_get_real_time` gives in this regard. In general, rough time synchronisation has to be guaranteed by the deployment, but IMO this can be expected of users (also these are second timestamps, and the app server can set the leeway via the expiry period). Maybe this should be mentioned in the documentation.

Timezones are another issue; i (probably wrongfully) assumed that the time is in the UTC timezone. Afaik setting up your servers with UTC time is a best practice in general, but if there is a clean API for getting guaranteed UTC timestamps then I would apply it here.

Reply:
This migth be in UTC already, at least looking at the documentation for `clock_gettime` and `CLOCK_REALTIME`, which is what `janus_get_real_time` wraps: https://linux.die.net/man/3/clock_gettime

    Its time represents seconds and nanoseconds since the Epoch

Anyway, not sure that's indeed the case.

Reply:
According to this stackoverflow it is:
https://stackoverflow.com/questions/8531845/is-clock-gettime-a-utc-time-or-local-time-zone